lib/api/environment.rb

```diff
@@ -1,7 +1,30 @@
 module Api
   class Environment
     def self.normalized_attributes
-      @normalized_attributes ||= {:time => {}, :url => {}, :resource => {}, :encrypted => {}}
+      @normalized_attributes ||= {
+        :time      => time_attributes.each_with_object({}) { |attr, hsh| hsh[attr] = true },
```

Comment:
@Fryguy I thought this too, but when I benchmarked it was consistently slower (compared with SortedSet too). Nonetheless, I'll put a PR together so perhaps we can discuss this better there